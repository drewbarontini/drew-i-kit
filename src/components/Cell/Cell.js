import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

import { setProp, theme } from '../../lib/utils';

const Cell = styled.div`
  ${setProp({
    prop: 'size',
    cssProp: 'max-width',
    themeKey: 'sizing',
    fallback: props => theme('sizing.base')(props),
  })};
  ${setProp({
    prop: 'minWidth',
    themeKey: 'sizing',
  })};

  ${ifProp(
    'center',
    css`
      margin-left: auto;
      margin-right: auto;
    `
  )};
`;

Cell.propTypes = {
  center: PropTypes.bool,
  minWidth: PropTypes.string,
  size: PropTypes.string,
};

Cell.displayName = 'Cell';

export default Cell;
