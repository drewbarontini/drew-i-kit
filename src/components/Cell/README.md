# Cell

## Usage

```jsx
import { Cell } from 'drew-i-kit';

<Cell>
  <p>Some content here...</p>
</Cell>
```

```jsx
import { Cell } from 'drew-i-kit';

<Cell center>
  <p>Some content here...</p>
</Cell>
<Cell size="s">
  <p>Some content here...</p>
</Cell>
<Cell size="l">
  <p>Some content here...</p>
</Cell>
```

## Props

- `center` (**Boolean**, optional)
- `minWidth` (**string**, optional)
- `size` (**string**, optional)