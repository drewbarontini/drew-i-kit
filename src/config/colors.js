import { paletteFor } from '../lib';

const colors = {
  ...paletteFor({
    area: 'primary',
    base: '#3576a7',
    darker: ['#152f43', '#1d415c', '#255375', '#2d648e'],
    lighter: ['#538bb4', '#729fc1', '#90B4cf', '#aeC8dc'],
  }),
  ...paletteFor({
    area: 'neutral',
    base: '#737373',
    darker: ['#0b0b0b', '#1d1d1d', '#2e2e2e', '#3f3f3f', '#515151', '#626262'],
    lighter: ['#888888', '#9d9d9d', '#b2b2b2', '#c7c7c7', '#dcdcdc', '#f1f1f1'],
  }),
  accent: {
    ...paletteFor({
      area: 'green',
      base: '#8acb88',
      darker: ['#233322', '#375136', '#4c704b', '#618e5f', '#75ad74'],
      lighter: ['#9cd39a', '#addbac', '#bfe2be', '#d0eacf', '#e2f2e1'],
    }),
    ...paletteFor({
      area: 'yellow',
      base: '#ffc527',
      darker: ['#40310a', '#664f10', '#8c6c15', '#b38a1b', '#d9a721'],
      lighter: ['#ffce47', '#ffd668', '#ffdf88', '#ffe8a9', '#fff1c9'],
    }),
    ...paletteFor({
      area: 'red',
      base: '#fe5f55',
      darker: ['#401815', '#662622', '#8c342f', '#b2433b', '#d85148'],
      lighter: ['#fe776f', '#fe8f88', '#fea7a2', '#ffbfbb', '#ffd7d5'],
    }),
  },
  bg: '#fff',
  fg: '#272727',
  white: '#fff',
  black: '#272727',
  border: '#dcdcdc',
};

export default colors;
