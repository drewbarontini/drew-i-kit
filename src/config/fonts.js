const fonts = {
  family: {
    base:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    heading:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    monospace: "'Monaco', monospace",
  },
  lineHeight: {
    base: '1.5',
    heading: '1.2',
  },
  sizes: {
    base: '14px',
    xxs: '8px',
    xs: '10px',
    s: '12px',
    l: '16px',
    xl: '24px',
    h1: '30px',
    h2: '26px',
    h3: '22px',
    h4: '18px',
  },
  weights: {
    light: 200,
    normal: 400,
    semiBold: 600,
    bold: 800,
  },
};

export default fonts;
