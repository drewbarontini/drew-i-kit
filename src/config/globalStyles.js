import { css } from 'styled-components';

import { palette, theme } from '../lib/props';
import { responsive } from '../lib/responsive';

const globalStyles = css`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  body {
    background-color: ${palette('bg')};
    color: ${palette('fg')};
    font-family: ${theme('fonts.family.base')};
    font-size: ${theme('fonts.sizes.base')};
    line-height: ${theme('fonts.lineHeight.base')};
    margin: ${theme('spacing.l')};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${theme('fonts.family.heading')};
    line-height: ${theme('fonts.lineHeight.heading')};
    margin: 0;
    margin-bottom: ${theme('spacing.s')};
  }

  img {
    display: block;
    max-width: 100%;
  }

  ${responsive};
`;

export default globalStyles;
