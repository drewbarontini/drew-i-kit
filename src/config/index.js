export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
