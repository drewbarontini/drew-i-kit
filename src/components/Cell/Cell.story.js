import React from 'react';
import { storiesOf } from '@storybook/react';

import Cell from './';
import Text from '../Text';

const content = `
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
  tempor incididunt ut labore et dolore magna aliqua. Mi bibendum neque
  egestas congue quisque egestas diam in. Massa tempor nec feugiat nisl
  pretium. Magnis dis parturient montes nascetur ridiculus mus. Nibh ipsum
  consequat nisl vel pretium lectus quam id leo. Ipsum nunc aliquet
  bibendum enim facilisis gravida neque convallis. Varius vel pharetra vel
  turpis nunc eget. Ut venenatis tellus in metus vulputate. Cras fermentum
  odio eu feugiat pretium nibh ipsum consequat nisl. Felis bibendum ut
  tristique et egestas quis. At consectetur lorem donec massa sapien
  faucibus et. Arcu non odio euismod lacinia at quis. Viverra vitae congue
  eu consequat ac felis donec et odio. Hac habitasse platea dictumst
  vestibulum rhoncus est pellentesque elit ullamcorper. Nisl vel pretium
  lectus quam id leo. Habitasse platea dictumst vestibulum rhoncus est
  pellentesque elit. Sit amet porttitor eget dolor morbi non. A diam
  sollicitudin tempor id eu nisl nunc mi. In nulla posuere sollicitudin
  aliquam ultrices.
`;

storiesOf('Cell', module)
  .add('default', () => (
    <Cell>
      <Text>{content}</Text>
    </Cell>
  ))
  .add('center', () => (
    <Cell center>
      <Text>{content}</Text>
    </Cell>
  ))
  .add('small', () => (
    <Cell size="s">
      <Text>{content}</Text>
    </Cell>
  ))
  .add('large', () => (
    <Cell size="l">
      <Text>{content}</Text>
    </Cell>
  ));
