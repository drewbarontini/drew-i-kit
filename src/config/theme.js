import colors from './colors';
import fonts from './fonts';
import icons from './icons';

const theme = {
  borders: {
    sizes: {
      base: '1px',
      l: '2px',
    },
    radius: {
      base: '3px',
    },
  },
  breakpoints: {
    sizes: {
      s: 380,
      m: 768,
      l: 900,
      xl: 1100,
    },
    query: 'max-width',
    unit: 'px',
  },
  colors: {
    ...colors,
  },
  fonts: {
    ...fonts,
  },
  icons: {
    ...icons,
  },
  shadows: {
    base: '0 3px 3px rgba(0, 0, 0, 0.25);',
  },
  sizing: {
    base: '900px',
    s: '400px',
    l: '1200px',
  },
  spacing: {
    base: '20px',
    xs: '5px',
    s: '10px',
    l: '40px',
    xl: '60px',
  },
  transition: {
    base: '0.3s ease-in-out',
  },
};

export default theme;
