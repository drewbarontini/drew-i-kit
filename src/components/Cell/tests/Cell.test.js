import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import Cell from '../';

describe('<Cell />', () => {
  it('should render properly', () => {
    const { container } = render(<Cell theme={theme}>My Cell</Cell>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.tagName.toLowerCase()).toEqual('div');
    expect(container.firstChild).toHaveTextContent('My Cell');
    expect(container.firstChild).toHaveStyleRule(
      'max-width',
      theme.sizing.base
    );
  });
});
