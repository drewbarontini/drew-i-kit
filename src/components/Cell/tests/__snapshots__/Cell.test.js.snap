// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> should render properly 1`] = `
.c0 {
  max-width: 900px;
}

<div
  class="c0"
>
  My Cell
</div>
`;
